### ISSUE TITLE
binary executables should use '.exe' suffix on Windows

### ISSUE THREAD

**Issue by 2bndy5** [cpp-linter/clang-tools-pip#11]
I've been trying this out on my Windows machine, and The OS didn't know that the binaries were executable files because they didn't have the `.exe` file suffix.

## Example after installing v12 in a python venv
<details><summary>Executing `clang-format-12` asked me what program to open the file with.</summary>

![image](https://user-images.githubusercontent.com/14963867/183004076-706f2225-8d3e-4f4e-b0b2-32b95ac0804e.png)
</details>

After changing `.exe` to the clang-format binary file's suffix, I got the expected behavior:
```ps
(venv) PS path\to\GitHub\cpp-linter-action> clang-format-12.exe --version
clang-format version 12.0.0
```
I'm guessing the fix for this would involve modifying the call to https://github.com/cpp-linter/clang-tools-pip/blob/4b4964444cd8afc0c13a3595692fdc7c4b8d06b7/clang_tools/install.py#L66
or simply adjusting the `new_file_name` arg for the Windows platform within the `mv_and_chmod()` function.

### Other Windows considerations
[`os.chmod()`](https://docs.python.org/3/library/os.html#os.chmod) has limited support on Windows. The docs specifically note:
> Although Windows supports [chmod()](https://docs.python.org/3/library/os.html#os.chmod), you can only set the file’s read-only flag with it (via the stat.S_IWRITE and stat.S_IREAD constants or a corresponding integer value). All other bits are ignored.

I'm not entirely familiar with Unix permission's octal format, but using `0o755` didn't raise any exceptions.

---
ps - I also would like it if `clang-tools -i 12` installed 12.0.1 instead of the older 12.0.0

**Comment by shenxianpeng** [comment]
It's my first time installing binary on Windows and it failed... 

```PowerShell
PS C:\Users\xshen> clang-tools --install 13
Don't have permission to install clang-format-13 to c:\program files\python38.             Try to run with the appropriate permissions.
PS C:\Users\xshen>
```

**Comment by shenxianpeng** [comment]
From @2bndy5 in https://github.com/cpp-linter/cpp-linter-action/issues/78

### regarding clang-tools-pip
It might be a good idea to create the binaries using .exe suffix (see https://github.com/cpp-linter/clang-tools-pip/issues/11) and a link to the binary that doesn't use the version in the title.
```
(venv) PS path > clang-tools -i 12
(venv) PS path > ls venv/Scripts
clang-format-12.exe
clang-format.lnk
clang-tidy-12.exe
clang-tidy.lnk
```
Though, this might cause conflicts if the unspecified directory (`-d`) is always `sys.executable` and multiple versions are installed.

**Comment by shenxianpeng** [comment]
> that doesn't use the version in the title.

Do you mean to type `clang-format` or `clang-tidy` can execute the clang-format-12.exe or clang-tidy-12.exe?
If multiple versions are installed? which binary should be linked?

If unspecified deriectory(`-d`) it should be fine because the binary name is different.

```
PS C:\Users\xshen> clang-tools --install 13
Don't have permission to install clang-format-13 to c:\program files\python38.             Try to run with the appropriate permissions.
PS C:\Users\xshen> clang-tools --install 12
Don't have permission to install clang-format-12 to c:\program files\python38.             Try to run with the appropriate permissions.
```

**Comment by 2bndy5** [comment]
> Do you mean to type clang-format or clang-tidy can execute the clang-format-12.exe or clang-tidy-12.exe?

yes.

>If multiple versions are installed? which binary should be linked?

I imagine the last version installed. Does this CLI accept multiple versions (like `clang-tools -i 12 -i 13`)?

**Comment by 2bndy5** [comment]
> ```
> Don't have permission to install clang-format-13 to c:\program files\python38.             Try to run with the appropriate permissions.
> ```

I didn't get this problem because I run all my python stuff in a venv. This might be a good idea (about using a venv) to add to the readme.

**Comment by shenxianpeng** [comment]
> Does this CLI accept multiple versions (like clang-tools -i 12 -i 13)?

No. to install multiple versions need to run `clang-tool -i <version>` multiple times. 😃

Tested with venv it works. If possible it should be better to fix the permission problem when the user doesn't use venv.

**Comment by 2bndy5** [comment]
I'm not a fan of downloading and executing scripts that require admin privileges. I'd prefer users activate a venv for ensured security.

**Comment by shenxianpeng** [comment]
I totally agree. maybe I can install the binary to the user folder like `C:\Users\xshen\.clang-tools` etc.

**Comment by 2bndy5** [comment]
I like that as a default value. The user can specify anything they want if that isn't suitable.

**Comment by 2bndy5** [comment]
So, I've pushed my changes to the pro-revision branch.

I also added the creation of a symlink if the tool was not previously installed. Users can use the new `-f` switch to force the symlink to overwrite a pre-existing symlink.

I tried using a different default install path on Windows, but it wasn't automatically in the CI's `PATH` env var:
https://github.com/cpp-linter/clang-tools-pip/blob/afa5f57e862cf1530dc19c80e4311d69cc553425/clang_tools/install.py#L66-L68
Its commented out for now, until we find a more permissive path that is in the PATH env var.
